internal/controller/backend_security_policy.go

```diff
@@ -100,7 +100,10 @@ func (c *BackendSecurityPolicyController) rotateCredential(ctx context.Context,
 	case aigv1a1.BackendSecurityPolicyTypeAzureCredentials:
 		secretRef := bsp.Spec.AzureCredentials.ClientSecretRef
 		if secretRef == nil {
-			return ctrl.Result{}, fmt.Errorf("azure credentials secret reference is nil, namespace %s name %s", bsp.Namespace, bsp.Name)
+			return ctrl.Result{}, fmt.Errorf("azure credentials secret ref is nil, namespace %s name %s", bsp.Namespace, bsp.Name)
+		}
+		if secretRef.Namespace == nil {
+			return ctrl.Result{}, fmt.Errorf("azure credentials secret ref namespace is nil, namespace %s name %s", bsp.Namespace, bsp.Name)
```

Comment:
hmm not sure, is this in line with the other secret ref ? The API says this is optional and if not give, it should default to the namespace of BSP instead of error like this